lib/private/DB/QueryBuilder/QueryBuilder.php

```diff
@@ -253,48 +246,63 @@ public function execute() {
 				'exception' => $exception,
 			]);
 		}
+	}
 
-		$result = $this->queryBuilder->execute();
-		if (is_int($result)) {
-			return $result;
+	/**
+	 * Executes this query using the bound parameters and their types.
+	 *
+	 * Uses {@see Connection::executeQuery} for select statements and {@see Connection::executeUpdate}
+	 * for insert, update and delete statements.
+	 *
+	 * @return IResult|int
+	 */
+	public function execute(?IDBConnection $connection = null) {
+		try {
+			if ($this->getType() === \Doctrine\DBAL\Query\QueryBuilder::SELECT) {
+				return $this->executeQuery($connection);
+			} else {
+				return $this->executeStatement($connection);
+			}
+		} catch (DBALException $e) {
```

Comment:
`DBALException` does not exists here.
What are you trying to catch?

Reply:
```suggestion
		} catch (DbalException $e) {
```

Actually it’s only a case issue, there is a `use` statement for `DbalException`